trie/trie_test.go

```diff
@@ -254,166 +310,200 @@ func TestPutAndGet(t *testing.T) {
+		{key: []byte{0x01, 0x35, 0x46}, value: []byte("raccoon"), op: PUT},
+		{key: []byte{0x01, 0x35, 0x46, 0x77}, value: []byte("rat"), op: PUT},
+		{key: []byte{0x09, 0xd3}, value: []byte("noot"), op: DEL},
+		{key: []byte{0x09, 0xd3}, value: nil, op: GET},
 	}
 
-	leaf, err = trie.getLeaf(key3)
-	if leaf == nil {
-		t.Errorf("Fail to get key %x: nil leaf", key3)
-	} else if err != nil {
-		t.Errorf("Fail to get key %x: %s", key3, err.Error())
-	} else if !bytes.Equal(leaf.value, value3) {
-		t.Errorf("Fail to get key %x with value %x: got %x", key3, value3, val)
-	} else if !bytes.Equal(leaf.key, pk3) {
-		t.Errorf("Fail to get correct partial key %x: got %x", pk3, leaf.key)
+	runTests(t, trie, tests)
+}
+
+func TestDeleteFromBranch(t *testing.T) {
+	trie := newEmpty()
+
+	tests := []trieTest{
+		{key: []byte{0x06, 0x15, 0xfc}, value: []byte("noot"), op: PUT},
+		{key: []byte{0x06, 0x2b, 0xa9}, value: []byte("nootagain"), op: PUT},
+		{key: []byte{0x06, 0xaf, 0xb1}, value: []byte("odd"), op: PUT},
+		{key: []byte{0x06, 0xa3, 0xff}, value: []byte("stuff"), op: PUT},
+		{key: []byte{0x43, 0x21}, value: []byte("stuffagain"), op: PUT},
+		{key: []byte{0x06, 0x15, 0xfc}, value: []byte("noot"), op: GET},
+		{key: []byte{0x06, 0x2b, 0xa9}, value: []byte("nootagain"), op: GET},
+		{key: []byte{0x06, 0x15, 0xfc}, value: []byte("noot"), op: DEL},
+		{key: []byte{0x06, 0x15, 0xfc}, value: nil, op: GET},
+		{key: []byte{0x06, 0x2b, 0xa9}, value: []byte("nootagain"), op: GET},
+		{key: []byte{0x06, 0xaf, 0xb1}, value: []byte("odd"), op: GET},
+		{key: []byte{0x06, 0xaf, 0xb1}, value: []byte("odd"), op: DEL},
+		{key: []byte{0x06, 0x2b, 0xa9}, value: []byte("nootagain"), op: GET},
+		{key: []byte{0x06, 0xa3, 0xff}, value: []byte("stuff"), op: GET},
+		{key: []byte{0x06, 0xa3, 0xff}, value: []byte("stuff"), op: DEL},
+		{key: []byte{0x06, 0x2b, 0xa9}, value: []byte("nootagain"), op: GET},
+	}
+
+	runTests(t, trie, tests)
+}
+
+func TestDeleteOddKeyLengths(t *testing.T) {
+	trie := newEmpty()
+
+	tests := []trieTest{
+		{key: []byte{0x43, 0xc1}, value: []byte("noot"), op: PUT},
+		{key: []byte{0x43, 0xc1}, value: []byte("noot"), op: GET},
+		{key: []byte{0x49, 0x29}, value: []byte("nootagain"), op: PUT},
+		{key: []byte{0x49, 0x29}, value: []byte("nootagain"), op: GET},
+		{key: []byte{0x43, 0x0c}, value: []byte("odd"), op: PUT},
+		{key: []byte{0x43, 0x0c}, value: []byte("odd"), op: GET},
+		{key: []byte{0x4f, 0x4d}, value: []byte("stuff"), op: PUT},
+		{key: []byte{0x4f, 0x4d}, value: []byte("stuff"), op: GET},
+		{key: []byte{0x43, 0x0c}, value: []byte("odd"), op: DEL},
+		{key: []byte{0x43, 0x0c}, value: nil, op: GET},
+		{key: []byte{0xf4, 0xbc}, value: []byte("spaghetti"), op: PUT},
+		{key: []byte{0xf4, 0xbc}, value: []byte("spaghetti"), op: GET},
+		{key: []byte{0x4f, 0x4d}, value: []byte("stuff"), op: GET},
+		{key: []byte{0x43, 0xc1}, value: []byte("noot"), op: GET},
+	}
+
+	runTests(t, trie, tests)
+}
+
+func TestDelete(t *testing.T) {
```

Comment:
whats unique about these tests such that they can't be abstracted like the rest?

Reply:
they are random